### PR TITLE
[embedded] Port and start building Darwin.swiftmodule as embedded

### PR DESCRIPTION
**Pull request by kubamracek**
Building in embedded Swift mode is useful even for non-standalone environments/targets, like Darwin, at least for testing but also possibly for writing userspace library code that wants to get the no-runtime no-metadata guarantees on embedded Swift. Let's make a step towards that by porting and providing an embedded build of the Darwin overlay (Darwin.swiftmodule).

### CODE REVIEW

**Comment by rauhul** [comment]
Is there a way I can download this toolchain built in CI and try it locally?

**Comment by kubamracek** [comment]
@rauhul  I think the swift-ci command I just made should produce a downloadable toolchain.

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please build toolchain macOS platform

File: stdlib/public/core/CMakeLists.txt [hunk 1]
```diff
@@ -49,7 +49,7 @@ split_embedded_sources(
     NORMAL BridgeStorage.swift
     NORMAL BridgingBuffer.swift
   EMBEDDED Builtin.swift
-    NORMAL BuiltinMath.swift
+  EMBEDDED BuiltinMath.swift
```
Comment (phausler) [inline stdlib/public/core/CMakeLists.txt:52]: This brings in BuiltinMath for all embedded? is that ok? (id say sure... I guess...)

**Review by phausler — APPROVED** [review]

File: test/embedded/darwin-bridging-header.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// RUN: %target-clang %t/main.o -o %t/a.out -dead_strip
+// RUN: %target-run %t/a.out | %FileCheck %s
+
+// REQUIRES: swift_in_compiler
+// REQUIRES: executable_test
+// REQUIRES: optimized_stdlib
+// REQUIRES: VENDOR=apple
+// REQUIRES: OS=macosx
+
+// BEGIN BridgingHeader.h
```
[the remark this answers is not in the document]
Comment (kubamracek) [reply test/embedded/darwin-bridging-header.swift:14]: It's `split_file.py` that's splitting the file based on these markers.

File: stdlib/public/core/CMakeLists.txt (hunk 1 above)
Comment (rauhul) [reply stdlib/public/core/CMakeLists.txt:52]: I feel like we should eventually conditionalize these + float on something like `hasFeature(HardFloat)` || hasFeature(SoftFloat)`. @stephentyrone do you have any thoughts on this?

**Pull request merged**